### PR TITLE
Prepare observations for the updates to the ensemble members.

### PR DESCRIPTION
**Pull request by jiaruidong2017**
# Description
This PR adds in initial configurations files and jobs to prepare the snow observations for the JEDI based snow DA updates to the EnkF ensemble members. Because IMS observations are background dependent for each ensemble member, we plan to put snow observations in the ensemble directory for the later snow analysis. 

Changes to the config.resources for the new task (add epsn)
Changes to the workflow tasks for the new task (add epsn)
Addition of new config file for snow DA related options (add config.epsn)

Because the snow observations are fetched from the COM_OBS in the gdas deterministic analysis and are saved into the COM_OBS in the enkfgdas analysis for the enkfgdas run, we have to define two different COM_OBS directories (COMIN_OBS vs COMOUT_OBS) and two obs basename heads (OPREFIX_IN vs OPREFIX_OUT) to identify the enkf analysis from the gdas deterministic analysis.  The MEMDIR is added to COM_OBS to identify among the different enkf members. Other minor changes include to add CASE in the IMS filename. 

This PR depends on https://github.com/NOAA-EMC/GDASApp/pull/998.

# Type of change
- New feature (adds jobs to prepare the snow obs for the enkf analysis)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO

# How has this been tested?
Hybrid snow DA test on Hera was successful

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Comment by WalterKolczynski-NOAA** [comment]
We'll need to wait for #2421 to be merged to test, since the relevant tests do not currently run on Orion and the CI system is set up to run on Rocky on Hera.

**Comment by jiaruidong2017** [comment]
> We'll need to wait for #2421 to be merged to test, since the relevant tests do not currently run on Orion and the CI system is set up to run on Rocky on Hera.

@WalterKolczynski-NOAA Please hold the CI test until the merge of https://github.com/NOAA-EMC/GDASApp/pull/998 and the updates of the commit hash from GDASApp. Thanks.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [inline jobs/JGLOBAL_PREP_SNOW_OBS]: @aerorahul @WalterKolczynski-NOAA are you both okay with these additional variables? Something about this I don't particularly like but I don't have a good suggestion right now either.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [inline jobs/JGLOBAL_PREP_SNOW_OBS:15]: what about `enkfgfs`?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [inline jobs/JGLOBAL_PREP_SNOW_OBS:20]: so we don't export oCDUMP if the run if gdas?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [inline jobs/JGLOBAL_PREP_SNOW_OBS]: shouldn't this be GDUMP?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [inline jobs/JGLOBAL_PREP_SNOW_OBS]: do we know for sure COM_OBS_IN exists already?

File: parm/config/gfs/config.com
Comment (CoryMartin-NOAA) [inline parm/config/gfs/config.com:41]: This is going to do this for all ob types, isn't it, is this okay @aerorahul ?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS:15]: I have no idea about `enkfgfs`.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS:20]: Do we need export `oCDUMP` here? Both rCDUMP and oCDUMP are used just locally.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS]: The `rCDUMP=${RUN}` is defined locally for the GFS cycle to get the restart conditions from GDAS. For the GDAS cycle, rCDUMP is the same as GDUMP.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS]: Yes, it is created in the `gdasprep` job.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (CoryMartin-NOAA) [reply jobs/JGLOBAL_PREP_SNOW_OBS:20]: sorry I misread this line, but if they are only used internally then why export rCDUMP above on line 13?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (WalterKolczynski-NOAA) [reply jobs/JGLOBAL_PREP_SNOW_OBS]: There are a couple things I don't like here, the main one being this will not be acceptable to operations where `COM_OBS_TMPL` points to the obsproc space and should be strictly input for GFS. The out needs to use a different template (probably a new `COM_SNOW_PREP_TMPL`).

Also, if IN and OUT are going to be used in the variable name, would prefer to have it be `COMIN_*` and `COMOUT_*`. That is the pattern we plan on using when we add IN and OUT to the rest of the workflow.

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS:20]: I reused this by following the old version `jobs/JGLOBAL_FORECAST` for the GFS cycle to get the restart conditions. But I just checked and found they were modified as below: 

```
# Restart conditions for GFS cycle come from GDAS
rCDUMP=${RUN}
export rCDUMP="${RUN/gfs/gdas}"
```

Do I need to make the corresponding changes?

File: jobs/JGLOBAL_PREP_SNOW_OBS
Comment (jiaruidong2017) [reply jobs/JGLOBAL_PREP_SNOW_OBS]: Thanks for the good suggestions, and I have made changes from `COM_OBS_IN/OUT` to `COMIN_OBS` and `COMOUT_OBS`. I have no idea and wait for the suggestions on using a new `COM_SNOW_PREP_TMPL` or not.

**Comment by jiaruidong2017** [comment]
@WalterKolczynski-NOAA I have updated the GDASApp to the current commit hash including the required changes by this PR. This PR is ready to proceed for a review. Thanks.

File: parm/config/gfs/config.com
Comment (CoryMartin-NOAA) [reply parm/config/gfs/config.com:41]: Pinging @aerorahul again. Are we okay with creating a `$MEMDIR` for all observations? Will this work as intended out of the box for the atmosphere @jiaruidong2017 ? I don't know if it is preferred to have 80 largely identical sets of directories or 1 directory for most obs and 80 directories with just snow in them?

Comment (jiaruidong2017) [reply parm/config/gfs/config.com:41]: If we want that the `COM_OBS` points only to the `gdas` directory and without `memdir`, we can simply add `MEMDIR=""` where the `COM_OBS` is defined in all other jobs.  An example is below: 

`RUN=${CDUMP} MEMDIR="" YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COM_OBS:COM_OBS_TMPL
`

**Comment by WalterKolczynski-NOAA** [comment]
@jiaruidong2017 Sorry, I missed that this one was ready and kept skipping over it because of the blocked label. Please merge in develop and resolve any conflicts and we'll run it through CI.

**Comment by CoryMartin-NOAA** [comment]
Hold off on that, I need to chat with @jiaruidong2017 but I think we are going to end up closing this PR due to a proposed science change.

**Comment by jiaruidong2017** [comment]
@WalterKolczynski-NOAA Because the GDASApp's PR was reverted, please hold up this PR until the GDASApp's PR is re-merged. Thanks.

**Comment by CoryMartin-NOAA** [comment]
We are going to try another direction for the ensemble update, and I'll close this PR.

**Pull request closed without merging**
